packages/@aws-cdk/aws-batch/lib/ecs-container-definition.ts

```diff
@@ -0,0 +1,949 @@
+   *
+   * @see https://docs.aws.amazon.com/efs/latest/ug/efs-mount-helper.html
+   *
+   * @default - chosen by the EFS Mount Helper
+   */
+  public readonly transitEncryptionPort?: number;
+
+  /**
+   * The Amazon EFS access point ID to use.
+   * If an access point is specified, `rootDirectory` must either be omitted or set to `/`
+   * which enforces the path set on the EFS access point.
+   * If an access point is used, `enableTransitEncryption` must be `true`.
+   *
+   * @see https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html
+   *
+   * @default - no accessPointId
+   */
+  public readonly accessPointId?: string;
+
+  /**
+   * Whether or not to use the AWS Batch job IAM role defined in a job definition when mounting the Amazon EFS file system.
+   * If specified, `enableTransitEncryption` must be `true`.
+   *
+   * @see https://docs.aws.amazon.com/batch/latest/userguide/efs-volumes.html#efs-volume-accesspoints
+   *
+   * @default false
+   */
+  public readonly useJobDefinitionRole?: boolean;
+
+  constructor(options: EfsVolumeOptions) {
+    super(options);
+
+    this.fileSystem = options.fileSystem;
+    this.rootDirectory = options.rootDirectory;
+    this.enableTransitEncryption = options.enableTransitEncryption;
+    this.transitEncryptionPort = options.transitEncryptionPort;
+    this.accessPointId = options.accessPointId;
+    this.useJobDefinitionRole = options.useJobDefinitionRole;
+  }
+}
+
+Object.defineProperty(EfsVolume.prototype, EFS_VOLUME_SYMBOL, {
+  value: true,
+  enumerable: false,
+  writable: false,
+});
+
+/**
+ * Options for configuring an ECS HostVolume
+ */
+export interface HostVolumeOptions extends EcsVolumeOptions {
+  /**
+   * The path on the host machine this container will have access to
+   *
+   * @default - Docker will choose the host path.
+   * The data may not persist after the containers that use it stop running.
+   */
+  readonly hostPath?: string;
+}
+/**
+ * Creates a Host volume. This volume will persist on the host at the specified `hostPath`.
+ * If the `hostPath` is not specified, Docker will choose the host path. In this case,
+ * the data may not persist after the containers that use it stop running.
+ */
+export class HostVolume extends EcsVolume {
+  /**
+   * returns `true` if `x` is a `HostVolume`, `false` otherwise
+   */
+  public static isHostVolume(x: any): x is HostVolume {
+    return x !== null && typeof (x) === 'object' && HOST_VOLUME_SYMBOL in x;
+  }
+
+  /**
+   * The path on the host machine this container will have access to
+   */
+  public readonly hostPath?: string;
+
+  constructor(options: HostVolumeOptions) {
+    super(options);
+    this.hostPath = options.hostPath;
+  }
+}
+
+Object.defineProperty(HostVolume.prototype, HOST_VOLUME_SYMBOL, {
+  value: true,
+  enumerable: false,
+  writable: false,
+});
+
+/**
+ * A container that can be run with ECS orchestration
+ */
+export interface IEcsContainerDefinition extends IConstruct {
```

Comment:
Is a containerdefinition a construct? Does it live on its own in the construct tree somewhere?

Reply:
I suppose there's also nothing AGAINST containerdefinition being a construct. Oh hell, why not.